flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/vector/VectorizedColumnBatch.java

```diff
@@ -132,4 +133,17 @@ public Decimal getDecimal(int rowId, int colId, int precision, int scale) {
 			return Decimal.fromUnscaledBytes(precision, scale, bytes);
 		}
 	}
+
+	public SqlTimestamp getTimestamp(int rowId, int colId, int precision) {
+		if (isNullAt(rowId, colId)) {
+			return null;
+		}
+
+		if (columns[colId] instanceof TimestampColumnVector) {
+			return ((TimestampColumnVector) (columns[colId])).getTimestamp(rowId, precision);
+		} else {
+			// by default, we assume the underlying LongColumnVector holds millisecond since Epoch.
+			return SqlTimestamp.fromEpochMillis(((LongColumnVector) columns[colId]).getLong(rowId));
```

Comment:
-1 for this implementation.
This implementation is not compatible with anyone. And now there is an interface(`TimestampColumnVector`), why should there be a specific implementation?